code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -384,8 +403,10 @@ nobiliumsuppression = INFINITY
 	var/temperature = air.temperature
 	var/turf/open/location = isturf(holder) ? holder : null
 	var/consumed_fuel = 0
-	if(temperature > 150 && temperature < T0C && cached_gases[/datum/gas/nitrogen][MOLES] > cached_gases[/datum/gas/hydrogen][MOLES])
+	if(cached_gases[/datum/gas/nitrogen][MOLES] > cached_gases[/datum/gas/hydrogen][MOLES])
 		consumed_fuel = (cached_gases[/datum/gas/hydrogen][MOLES] / AMMONIA_FORMATION_FACTOR)
+		if(cached_gases[/datum/gas/nitrogen][MOLES] - consumed_fuel < 0 || cached_gases[/datum/gas/hydrogen][MOLES] - consumed_fuel * 4 < 0)
+			return NO_REACTION
```

Comment:
well yes, but this check is to prevent the reaction making negative mole count and we don't want that

Reply:
mmmmm.

Reactions code should be able to prevent negative molar counts, at least if it's doing its job right.

Up to you, if it restricts the behavior too much feel free to remove it.

Reply:
It shouldn't even be noticeable since either you are pumping in gases or the reaction is close to finish. Either way this should be keeped as a safeguard